File: fx/src/main/java/gargoyle/fx/FXFilePreferencesFactory.java
package gargoyle.fx;

import java.util.prefs.Preferences;
import java.util.prefs.PreferencesFactory;

public final class FXFilePreferencesFactory implements PreferencesFactory {
    public FXFilePreferencesFactory() {
    }

    @Override
    public Preferences systemRoot() {
        return FXFilePreferences.getSystemRoot();
    }

    @Override
    public Preferences userRoot() {
        return FXFilePreferences.getUserRoot();
    }
}

File: fx/src/main/kotlin/gargoyle/fx/FxFilePreferencesFactory.kt
package gargoyle.fx

import java.util.prefs.Preferences
import java.util.prefs.PreferencesFactory

class FxFilePreferencesFactory : PreferencesFactory {
    override fun systemRoot(): Preferences =
        FxFilePreferences.systemRoot.value

    override fun userRoot(): Preferences =
        FxFilePreferences.userRoot.value
}
